nix/shell.sh

```diff
@@ -27,38 +28,51 @@ if ! command -v "nix" >/dev/null 2>&1; then
   fi
 fi
 
-if command -v "nix" >/dev/null 2>&1; then
-  platform=${TARGET_OS:=all}
-  if [ "$platform" != 'all' ] && [ "$platform" != 'android' ] && [ "$platform" != 'ios' ]; then
-    # This is a dirty workaround to the fact that 'yarn install' is an impure operation, so we need to call it from an impure shell. Hopefull we'll be able to fix this later on with something like yarn2nix
-    nix-shell --show-trace --argstr target-os ${TARGET_OS} --run "scripts/prepare-for-desktop-platform.sh" || exit
-  fi
+if !command -v "nix" >/dev/null 2>&1; then
+  echo "Nix not available, sourcing profile failed!"
+  exit 1
+fi
 
-  entry_point='shell.nix'
-  attrset_option=''
-  if [ -n "${_NIX_ATTR}" ]; then
-    entry_point='default.nix'
-    attrset_option="-A ${_NIX_ATTR}"
-  fi
+shellArgs=(
+  "--show-trace"
+)
+
+targetArg=""
+if [[ -n "${TARGET_OS}" ]]; then
```

Comment:
http://mywiki.wooledge.org/BashFAQ/031

>To cut a long story short: test implements the old, portable syntax of the command. In almost all shells (the oldest Bourne shells are the exception), [ is a synonym for test (but requires a final argument of ]). Although all modern shells have built-in implementations of [, there usually still is an external executable of that name, e.g. /bin/[. POSIX defines a mandatory feature set for [, but almost every shell offers extensions to it. So, if you want portable code, you should be careful not to use any of those extensions.

>[[ is a new, improved version of it, and it is a keyword rather than a program. This makes it easier to use, as shown below. [[ is understood by KornShell, Zsh and BASH (as of version 2.02), but not by other POSIX shell implementations (e.g. posh, yash, or dash) or the BourneShell.

If we are using bash, `[[ ]]` is the preferred way.

Reply:
Wasn't aware of that, I was using `[[` only when absolutely needed.